rust/operator/src/operations/pdb.rs

```diff
@@ -0,0 +1,143 @@
+use std::cmp::{max, min};
+
+use snafu::{ResultExt, Snafu};
+use stackable_hdfs_crd::{constants::APP_NAME, HdfsCluster, HdfsRole};
+use stackable_operator::{
+    builder::pdb::PodDisruptionBudgetBuilder, client::Client, cluster_resources::ClusterResources,
+    commons::pdb::PdbConfig, kube::ResourceExt,
+};
+
+use crate::{hdfs_controller::RESOURCE_MANAGER_HDFS_CONTROLLER, OPERATOR_NAME};
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("Cannot create PodDisruptionBudget for role group [{role_group}]"))]
+    CreatePdb {
+        source: stackable_operator::error::Error,
+        role_group: String,
+    },
+    #[snafu(display("Cannot apply role group PodDisruptionBudget [{name}]"))]
+    ApplyPdb {
+        source: stackable_operator::error::Error,
+        name: String,
+    },
+}
+
+pub async fn add_pdbs(
+    pdb: &PdbConfig,
+    hdfs: &HdfsCluster,
+    role: &HdfsRole,
+    client: &Client,
+    cluster_resources: &mut ClusterResources,
+) -> Result<(), Error> {
+    if !pdb.enabled {
+        return Ok(());
+    }
+    let max_unavailable = pdb.max_unavailable.unwrap_or(match role {
+        HdfsRole::NameNode => max_unavailable_name_nodes(),
+        HdfsRole::DataNode => max_unavailable_data_nodes(
+            hdfs.num_datanodes(),
+            hdfs.spec.cluster_config.dfs_replication as u16,
+        ),
+        HdfsRole::JournalNode => max_unavailable_journal_nodes(),
+    });
+    let pdb = PodDisruptionBudgetBuilder::new_with_role(
+        hdfs,
+        APP_NAME,
+        &role.to_string(),
+        OPERATOR_NAME,
+        RESOURCE_MANAGER_HDFS_CONTROLLER,
+    )
+    .with_context(|_| CreatePdbSnafu {
+        role_group: role.to_string(),
+    })?
```

Comment:
The error states rolegroup, role is passed though?

Reply:
Good catch! Fixed it